tests/filecheck/dialects/fsm/fsm_op.mlir

```diff
@@ -0,0 +1,146 @@
+// RUN XDSL_ROUNDTRIP
+
+// CHECK:  "fsm.machine"() ({
+// CHECK:  ^bb0(%arg0: i1):
+// CHECK:    "fsm.state"() ({
+// CHECK:    }, {
+// CHECK:    }) {sym_name = "IDLE"} : () -> ()
+// CHECK:  }) {function_type = (i1) -> (), initialState = "IDLE", sym_name = "foo"} : () -> ()
+
+"builtin.module"() ({
+  "fsm.machine"() ({
+  ^bb0(%arg0: i1):
+    "fsm.state"() ({
+    }, {
+    }) {sym_name = "IDLE"} : () -> ()
+  }) {function_type = (i1) -> (), initialState = "IDLE", sym_name = "foo"} : () -> ()
+}) : () -> ()
+
+
+// CHECK:  ^bb0(%arg0: i1):
+// CHECK:    %0 = "fsm.variable"() {initValue = 0 : i16, name = "cnt"} : () -> i16
+// CHECK:    "fsm.state"() ({
+// CHECK:      %1 = "arith.constant"() {value = true} : () -> i1
+// CHECK:      "fsm.output"(%1) : (i1) -> ()
+// CHECK:    }, {
+// CHECK:      "fsm.transition"() ({
+// CHECK:        "fsm.return"(%arg0) : (i1) -> ()
+// CHECK:      }, {
+// CHECK:        %1 = "arith.constant"() {value = 256 : i16} : () -> i16
+// CHECK:        "fsm.update"(%0, %1) : (i16, i16) -> ()
+// CHECK:      }) {nextState = @BUSY} : () -> ()
+// CHECK:    }) {sym_name = "IDLE"} : () -> ()
+// CHECK:    "fsm.state"() ({
+// CHECK:      %1 = "arith.constant"() {value = false} : () -> i1
+// CHECK:      "fsm.output"(%1) : (i1) -> ()
+// CHECK:    }, {
+// CHECK:      "fsm.transition"() ({
+// CHECK:        %1 = "arith.constant"() {value = 0 : i16} : () -> i16
+// CHECK:        %2 = "arith.cmpi"(%0, %1) {predicate = 1 : i64} : (i16, i16) -> i1
+// CHECK:        "fsm.return"(%2) : (i1) -> ()
+// CHECK:      }, {
+// CHECK:        %1 = "arith.constant"() {value = 1 : i16} : () -> i16
+// CHECK:        %2 = "arith.subi"(%0, %1) : (i16, i16) -> i16
+// CHECK:        "fsm.update"(%0, %2) : (i16, i16) -> ()
+// CHECK:      }) {nextState = @BUSY} : () -> ()
+// CHECK:      "fsm.transition"() ({
+// CHECK:        %1 = "arith.constant"() {value = 0 : i16} : () -> i16
+// CHECK:        %2 = "arith.cmpi"(%0, %1) {predicate = 0 : i64} : (i16, i16) -> i1
+// CHECK:        "fsm.return"(%2) : (i1) -> ()
+// CHECK:      }, {
+// CHECK:      }) {nextState = @IDLE} : () -> ()
+// CHECK:    }) {sym_name = "BUSY"} : () -> ()
+// CHECK:  }) {function_type = (i1) -> i1, initialState = "IDLE", sym_name = "foo"} : () -> ()
+
+
```

Comment:
Maybe this is my ignorance, but can you maybe describe in a short sentence what this FSM does?

Reply:
sure! great suggestion actually, I will do it

Reply:
@luisacicolini can you please address Josse's feedback with a very brief comment please?